packages/go/ecosystem/ecosystem.go

```diff
@@ -0,0 +1,64 @@
+package ecosystem
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/ossf/osv-schema/packages/go/constants"
+)
+
+// Parsed represents an ecosystem-with-suffix string as defined by the spec, parsed into
+// a structured format.
+//
+// The suffix is optional and is separated from the ecosystem by a colon.
+//
+// For example, "npm:abc" would be parsed into Parsed{Ecosystem: constants.EcosystemNPM, Suffix: "abc"}
```

Comment:
npm is not an ecosystem with suffix in spec. to avoid confusion, I think it's better to use some ecosystem allowing suffix.